test/testutil/test_main.c

```diff
@@ -9,14 +9,24 @@
 
 #include "../testutil.h"
 #include "output.h"
+#include "tu_local.h"
 
 #include <stdio.h>
+#include <stdlib.h>
 
 int test_main(int argc, char *argv[])
 {
+    int r;
+
     if (argc > 1)
         test_printf_stderr("Warning: ignoring extra command-line arguments.\n");
 
-    register_tests();
-    return run_tests(argv[0]);
+    if (!register_tests()) {
+        cleanup_tests();
+        test_verdict(0, "registration failed");
```

Comment:
I would simply remove this line.  From a TAP point of view, the output is wrong anyway, it should be a skip, not an "ok" / "not ok", and furthermore, the tests that still implement their own `test_main` don't output anything when they fail initialization.  Not outputting anything in this case is fine, 'cause the test recipes in `test/recipes` that wrap them will display their own "ok" / "not ok" depending on the exit code of the test program.